Assets/Tests/EditMode/Scripts/Storage/GlobalStorage_VM_Test.cs

```diff
@@ -0,0 +1,231 @@
+            Vector3 location2 = new Vector3(1, 1, 1);
+
+            GlobalStorage_VM.AddChest(chest, location);
+            GlobalStorage_VM.AddChest(chest2, location2);
+
+            Assert.AreEqual(chest2, GlobalStorage_VM.GetClosestChest(new Vector3(0.5f, 0.5f, 0.5f)));
+        }
+
+        [Test]
+        public void GlobalStorage_VM_GetItemCount_Pass()
+        {
+            GlobalStorage_VM.chests = new Dictionary<Chest_VM, Vector3>();
+            GlobalStorage_VM.itemLocations = new Dictionary<GameObject, Chest_VM>();
+            GameObject game = new GameObject();
+            game.AddComponent<Chest_VM>();
+            Chest_VM chest = game.GetComponent<Chest_VM>();
+            GameObject item = new GameObject();
+            item.name = "testItem";
+
+            chest.AddItem(item);
+            GlobalStorage_VM.chests.Add(chest, new Vector3(0, 0, 0));
+
+            Assert.AreEqual(1, GlobalStorage_VM.GetItemCount("testItem"));
+        }
+
+        [Test]
+        public void GlobalStorage_VM_GetChestWithItem_Pass()
+        {
+            GlobalStorage_VM.chests = new Dictionary<Chest_VM, Vector3>();
+            GlobalStorage_VM.itemLocations = new Dictionary<GameObject, Chest_VM>();
+            GameObject game = new GameObject();
+            game.AddComponent<Chest_VM>();
+            game.AddComponent<Item>();
+            Chest_VM chest = game.GetComponent<Chest_VM>();
+            Item item = game.GetComponent<Item>();
+            item.itemName = "testItem";
+            GameObject objItem = new GameObject();
+            objItem.name = "testItem";
+
+            chest.AddItem(objItem);
+            GlobalStorage_VM.chests.Add(chest, new Vector3(0, 0, 0));
+            List<Chest_VM> listOfChests = GlobalStorage_VM.GetChestWithItem(item);
+
+            Assert.AreEqual(chest, listOfChests[0]);
+        }
+
+        [Test]
+        public void GlobalStorage_VM_GetRandomChest_ExistingChest_Pass()
+        {
+            GlobalStorage_VM.chests = new Dictionary<Chest_VM, Vector3>();
+            GlobalStorage_VM.itemLocations = new Dictionary<GameObject, Chest_VM>();
+            GameObject game = new GameObject();
+            game.AddComponent<Chest_VM>();
+            Chest_VM chest = game.GetComponent<Chest_VM>();
+            Vector3 location = new Vector3(0, 0, 0);
+
+            GlobalStorage_VM.AddChest(chest, location);
+
+            Assert.AreEqual(chest, GlobalStorage_VM.GetRandomChest());
+        }
+
+        [Test]
+        public void GlobalStorage_VM_GetRandomChest_NoChest_Pass()
+        {
+            GlobalStorage_VM.chests = new Dictionary<Chest_VM, Vector3>();
+            GlobalStorage_VM.itemLocations = new Dictionary<GameObject, Chest_VM>();
+
+            if (GlobalStorage_VM.GetRandomChest() != null)
+            {
+                Assert.Fail();
+            }
+            else
+            {
+                Assert.Pass();
+            }
+        }
+    }
+}
```

Comment:
Great job on the unit tests and getting 100% coverage. For future unit tests, perhaps it might be helpful to use [Setup] to prevent having to rewrite code. If you're interested in learning more, I thought [Kodeco](https://www.kodeco.com/9454-introduction-to-unity-unit-testing#toc-anchor-012) had a great description in the "Setting Up and Tearing Down Phases" section.

Reply:
Noted. Will make sure to incorporate that into future tests. Thanks for the tips and feedback.